packages/common/keys/src/dxn.ts

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { invariant } from '@dxos/invariant';
+
+export class DXN {
+  /**
+   * Kind constants.
+   */
+  static kind = Object.freeze({
+    ECHO: 'echo',
+    TYPE: 'type',
+  });
+
+  static parse(dxn: string): DXN {
+    const [prefix, kind, ...parts] = dxn.split(':');
+    if (!(prefix === 'dxn')) {
+      throw new Error('Invalid DXN');
+    }
+    if (!(typeof kind === 'string' && kind.length > 0)) {
+      throw new Error('Invalid DXN');
+    }
+    if (!(parts.length > 0)) {
+      throw new Error('Invalid DXN');
+    }
+    return new DXN(kind, parts);
+  }
+
+  #kind: string;
```

Comment:
not seen this hash syntax before, what does this do?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties